src/App.tsx

```diff
@@ -1,61 +1,137 @@
 import './App.scss';
+import './field.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [correctTitle, setCorrectTitle] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [chosenUser, setChosenUser] = useState(false);
+
+  const newTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const toDoItem = {
+    id: newTodoId,
+    title,
+    userId,
+    completed: false,
+    user: getUser(userId),
+  };
+
+  const regex = /^[\p{L}\d\s]+$/u;
```

Comment:
(For future) try to move independent variables out of components to avoid it's creating on every re-rendering, it will  save some time